CMakeLists.txt

```diff
@@ -1,69 +1,160 @@
-cmake_minimum_required(VERSION 3.21)
+cmake_minimum_required(VERSION 3.24)
```

Comment:
Which addition requires this update?

Reply:
I can't remember exactly - something to do with FetchContent in test/CMakeLists.txt. 
Just tried with  3.21 and it breaks.

Reply:
Ubuntu 22.04 ships with cmake 3.22.1 which seems to work(?). I would rather restrict to 3.22 if possible.

Reply:
Are you finding googletest with the find_package? I just tried with 3.22 and that isn't working either - I might be doing something wrong?

Reply:
Yes - it downloads it at cmake time then builds at make time.

Reply:
If I set `set(FETCHCONTENT_QUIET OFF)`:

```
-- Build files have been written to: /home/js0259/git-ukaea/NESO-Particles/_deps/googletest-subbuild
[ 11%] Creating directories for 'googletest-populate'
[ 22%] Performing download step (download, verify and extract) for 'googletest-populate'
-- Downloading...
   dst='/home/js0259/git-ukaea/NESO-Particles/_deps/googletest-subbuild/googletest-populate-prefix/src/054a986a8513149e8374fc669a5fe40117ca6b41.zip'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/google/googletest/archive/054a986a8513149e8374fc669a5fe40117ca6b41.zip'
-- Downloading... done
-- extracting...
     src='/home/js0259/git-ukaea/NESO-Particles/_deps/googletest-subbuild/googletest-populate-prefix/src/054a986a8513149e8374fc669a5fe40117ca6b41.zip'
     dst='/home/js0259/git-ukaea/NESO-Particles/_deps/googletest-src'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[ 33%] No update step for 'googletest-populate'
[ 44%] No patch step for 'googletest-populate'
[ 55%] No configure step for 'googletest-populate'
[ 66%] No build step for 'googletest-populate'
[ 77%] No install step for 'googletest-populate'
[ 88%] No test step for 'googletest-populate'
[100%] Completed 'googletest-populate'
[100%] Built target googletest-populate
```

Reply:
Figured out the specific thing  it's this - 
https://cmake.org/cmake/help/latest/policy/CMP0135.html